pandas/core/indexes/multi.py

```diff
@@ -3475,6 +3475,8 @@ def intersection(self, other, sort=False):
         other, result_names = self._convert_can_do_setop(other)
 
         if self.equals(other):
+            if self.has_duplicates:
+                return self.unique()
```

Comment:
do we need result_names here?

Reply:
Yes, thanks very much. That came in via a pr after I did my changes